publishing/publish-operator.py

```diff
@@ -8,96 +19,113 @@
     if resp.returncode != 0:
-        print("Failed to set upstream/master")
+        print("failed to add upstream remote")
         sys.exit(1)
 
-    # community-operators
-    branch_name = 'add-community-hive-{}'.format(params.new_version)
-    pr_title = "Hive community operator {}".format(params.new_version)
-    open_pr(pr_title, personal_gh_user, branch_name, community_operator_repo_full_path, params.bundle_dir, COMMUNITY_OPERATOR_DIR, params.new_version)
-
-    # upstream-community-operators
-    branch_name = 'add-upstream-community-hive-{}'.format(params.new_version)
-    pr_title = "Hive upstream community operator {}".format(params.new_version)
-    open_pr(pr_title, personal_gh_user, branch_name, community_operator_repo_full_path, params.bundle_dir, UPSTREAM_COMMUNITY_OPERATOR_DIR, params.new_version)
-
-
-def open_pr(pr_title, gh_username, new_branch_name, repo_full_path, bundle_source_dir, bundle_target_dir_name, new_version):
-    print("Starting {}".format(pr_title))
+    print("Fetching latest upstream")
+    cmd = "git fetch upstream".split()
+    resp = subprocess.run(cmd, stdout=SUBPROCESS_REDIRECT)
+    if resp.returncode != 0:
+        print("failed to fetch upstream")
+        sys.exit(1)
 
     # Starting branch
-    cmd = "git checkout {}".format(COMMUNITY_OPERATOR_MAIN_BRANCH).split()
+    print("Checkout latest upstream/main")
+    cmd = "git checkout upstream/main".split()
     resp = subprocess.run(cmd, stdout=SUBPROCESS_REDIRECT, stderr=SUBPROCESS_REDIRECT)
     if resp.returncode != 0:
-        print("Failed to switch to {} branch".format(COMMUNITY_OPERATOR_MAIN_BRANCH))
+        print("Failed to switch to upstream/main branch")
         sys.exit(1)
 
-    print("Create branch {}".format(new_branch_name))
-    cmd = "git checkout -b {}".format(new_branch_name).split()
+    branch_name = 'update-hive-{}'.format(new_version)
+    pr_title = "Update Hive community operator to {}".format(new_version)
+    print("Starting {}".format(pr_title))
+
+    print("Create branch {}".format(branch_name))
+    cmd = "git checkout -b {}".format(branch_name).split()
     resp = subprocess.run(cmd, stdout=SUBPROCESS_REDIRECT, stderr=SUBPROCESS_REDIRECT)
     if resp.returncode != 0:
-        print("Faled to checkout branch {}".format(new_branch_name))
+        print("Failed to checkout branch {}".format(branch_name))
         sys.exit(1)
 
     # copy bundle directory
+    print("Copying bundle directory")
     bundle_files = os.path.join(bundle_source_dir, new_version)
-    hive_dir = os.path.join(repo_full_path, bundle_target_dir_name, new_version)
+    hive_dir = os.path.join(repo_full_path, HIVE_SUB_DIR, new_version)
     shutil.copytree(bundle_files, hive_dir)
 
     # update bundle manifest
-    bundle_manifests_file = os.path.join(repo_full_path, bundle_target_dir_name, "hive.package.yaml")
+    print("Updating bundle manfiest")
+    bundle_manifests_file = os.path.join(repo_full_path, HIVE_SUB_DIR, "hive.package.yaml")
```

Comment:
Unless you've changed the generate-operator-bundle-operatorhub.py script (or I'm missing something), this will result in all the OCM channels being removed from hive.package.yaml.

Reply:
It looks like it's there but it 2.4 should be updated to 1.1.13 as well https://github.com/k8s-operatorhub/community-operators/blob/main/operators/hive-operator/hive.package.yaml

I'll see how we can handle this

Reply:
The generate bundle script does not do this, but the publish operatorhub script does take this into account which is why the ocm versions were preserved. I've added a list param --update-channel for the channels you want to update. It is required and presumably you must always specify 'alpha', but can also add 'ocm-2.4'. I will update the hive monitor PR as well to document what to specify.